include/oneapi/mkl/detail/get_device_id.hpp

```diff
@@ -59,11 +59,19 @@ inline oneapi::mkl::device get_device_id(sycl::queue &queue) {
         else if (vendor_id == AMD_ID)
             device_id = device::amdgpu;
         else {
+#ifdef ENABLE_GENERIC_DEVICE
```

Comment:
It seems to me we could avoid these ifdef and always return a `generic_device` as a fallback. The `table_initializer` object should throw `backend_not_found` the `generic_device` is used with a backend that does not support it.